### PR TITLE
Climb

### PR DESCRIPTION
**Pull request by YHerm**

### CODE REVIEW

File: src/main/java/edu/greenblitz/robotName/OI.java [hunk 1]
```diff
@@ -165,12 +192,11 @@ public void fourthJoystickButtons() {
 	}
 
 	public void initializeDefaultCommands() {
-		Battery.getInstance().setDefaultCommand(new BatteryLimiter());
+//		Battery.getInstance().setDefaultCommand(new BatteryLimiter());
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/OI.java]: ...

File: src/main/java/edu/greenblitz/robotName/OI.java [hunk 2]
```diff
@@ -64,6 +69,7 @@ private OI() {
 		thirdJoystick = new SmartJoystick(RobotConstants.Joystick.THIRD);
 		fourthJoystick = new SmartJoystick(RobotConstants.Joystick.FOURTH);
 
+		thirdJoystickButtons();
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/OI.java]: remove, and add the needed buttons to schori

File: src/main/java/edu/greenblitz/robotName/ScoringModeSelector.java [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
+
 public class ScoringModeSelector {
 
 	private static ScoringMode currentMode;
 
 	public static void init() {
 		if (currentMode == null) {
 			currentMode = ScoringMode.SPEAKER;
+			ShuffleboardTab tab = Shuffleboard.getTab("auto");
+			tab.addString("Scoring Mode",  () -> currentMode.name());
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/ScoringModeSelector.java]: why here..?

File: src/main/java/edu/greenblitz/robotName/commands/arm/elbow/ElbowDefaultCommand.java [hunk 4]
```diff
@@ -9,6 +13,12 @@ public void initialize() {
 	
 	@Override
 	public void execute() {
-		elbow.standInPlace();
+		if (ScoringModeSelector.isClimbMode()) {
+			elbow.setPower(0);
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/arm/elbow/ElbowDefaultCommand.java:17]: i think we have stop()

File: src/main/java/edu/greenblitz/robotName/commands/climbing/PushSolenoidWheel.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+	@Override
+	public void initialize() {
+		timer.restart();
+	}
+
+	@Override
+	public void execute() {
+		lifter.setPower(LifterConstants.LIFTER_POWER_TO_RELEASES_SOLENOID);
+		if (timer.hasElapsed(SolenoidConstants.SECONDS_TO_RELEASE_SOLENOID / 2)) {
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/climbing/PushSolenoidWheel.java]: 2?

File: src/main/java/edu/greenblitz/robotName/commands/climbing/lifter/MoveLifterByJoystick.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+	public MoveLifterByJoystick(SmartJoystick joystick) {
+		super();
+		this.joystick = joystick;
+		solenoid = Solenoid.getInstance();
+	}
+
+	@Override
+	public void execute() {
+		lifter.setPower((joystick.getAxisValue(SmartJoystick.Axis.LEFT_Y) - 0.1) * (4 / 9.0));
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/climbing/lifter/MoveLifterByJoystick.java]: magic....

File: src/main/java/edu/greenblitz/robotName/commands/climbing/lifter/PushSolenoidAndLifter.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+	@Override
+	public void execute() {
+		solenoid.closeSolenoid();
+		lifter.setPower(LifterConstants.LIFTER_POWER_TO_RELEASES_SOLENOID);
+	}
+
+	@Override
+	public boolean isFinished() {
+		return timer.hasElapsed(SolenoidConstants.SECONDS_TO_RELEASE_SOLENOID / 2);
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/climbing/lifter/PushSolenoidAndLifter.java]: 2..?

File: src/main/java/edu/greenblitz/robotName/commands/climbing/lifter/PushSolenoidWheelOnly.java [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+	}
+
+	@Override
+	public void execute() {
+		lifter.setPower(LifterConstants.LIFTER_POWER_TO_RELEASES_SOLENOID);
+	}
+
+	@Override
+	public boolean isFinished() {
+		return timer.hasElapsed(SolenoidConstants.SECONDS_TO_RELEASE_SOLENOID / 2);
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/climbing/lifter/PushSolenoidWheelOnly.java]: 2..?

File: src/main/java/edu/greenblitz/robotName/commands/climbing/lifter/SlowRetractLifter.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+	public SlowRetractLifter() {
+		super();
+		timer = new Timer();
+	}
+
+	@Override
+	public void initialize() {
+		timer.restart();
+		startingPose = lifter.getPosition();
+		power = lifter.getMaxCurrent() > 25 ? LifterConstants.END_OF_CLIMB_POWER: LifterConstants.POWER_TO_RETRACT_LIFTER_SLOW;
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/climbing/lifter/SlowRetractLifter.java]: 25?

File: src/main/java/edu/greenblitz/robotName/commands/getNoteToSystem/TransferNote.java [hunk 10]
```diff
@@ -12,11 +14,11 @@ public class TransferNote extends SequentialCommandGroup {
 	public TransferNote() {
 		super(
 				new MoveToTransferNotePosition(),
-				new WaitCommand(0.5),
+				new WaitCommand(0.5).andThen(new InstantCommand(() -> SmartDashboard.putNumber("chahchah", 1))),
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/getNoteToSystem/TransferNote.java]: tf

File: src/main/java/edu/greenblitz/robotName/subsystems/climber/lifter/LifterConstants.java [hunk 11]
```diff
@@ -0,0 +1,44 @@
+package edu.greenblitz.robotName.subsystems.climber.lifter;
+
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.math.geometry.Translation3d;
+
+public class LifterConstants {
+
+    public static final double POWER_TO_EXTEND_LIFTER = 0.7;
+
+    public static final double POWER_TO_RETRACT_LIFTER = -0.6;
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/climber/lifter/LifterConstants.java:10]: add a space

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/commands/arm/elbow/ElbowDefaultCommand.java (hunk 4 above)
Comment (YoniKiriaty) [reply src/main/java/edu/greenblitz/robotName/commands/arm/elbow/ElbowDefaultCommand.java:17]: we dont.

**Pull request closed without merging**